### PR TITLE
ibcEvmTx (app) add new module to the Evmos app (app.go)

### PR DESCRIPTION
**Pull request by GAtom22**
## Description

Added the new module in the Evmos app struct and the corresponding initialization logic

Closes: #XXX

______

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.

PR review checkboxes:

I have...

- [ ] added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/evmos/evmos/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link in the PR description to the relevant issue or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all required CI checks have passed

Code maintenance:

I have...

- [ ] written unit and integration [tests](https://github.com/evmos/evmos/blob/main/CONTRIBUTING.md#testing)
- [ ] added relevant [`godoc`](https://go.dev/blog/godoc) and [code comments](https://blog.jbowen.dev/2019/09/the-magic-of-go-comments/).
- [ ] updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)

______

### Reviewers Checklist

**All** items are required. Please add a note if the item is not applicable and please add your handle next to the items reviewed if you only reviewed selected items.

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] confirmed that this PR does not change production code

### CODE REVIEW

**Review by Vvaradinov — APPROVED** [review]
LGTM!

File: app/app.go [hunk 1]
```diff
@@ -128,6 +128,12 @@ import (
 	erc20client "github.com/evmos/evmos/v10/x/erc20/client"
 	erc20keeper "github.com/evmos/evmos/v10/x/erc20/keeper"
 	erc20types "github.com/evmos/evmos/v10/x/erc20/types"
+	ibcevmtx "github.com/evmos/evmos/v10/x/ibc/evm"
```
Comment (danburck) [inline app/app.go:131]: nit, `tx` is not port of the path and it's nice to stay consistent with the path

```suggestion
	ibcevm "github.com/evmos/evmos/v10/x/ibc/evm"
```

File: app/app.go [hunk 2]
```diff
@@ -128,6 +128,12 @@ import (
...
+
+	// TODO when client is added
+	// ibcevmtxclient "github.com/evmos/evmos/v10/x/ibc/evm/client"
+	ibcevmtxkeeper "github.com/evmos/evmos/v10/x/ibc/evm/keeper"
```
Comment (danburck) [inline app/app.go:135]:
```suggestion
	ibcevmkeeper "github.com/evmos/evmos/v10/x/ibc/evm/keeper"
```

File: app/app.go [hunk 3]
```diff
@@ -128,6 +128,12 @@ import (
...
+	ibcevmtxtypes "github.com/evmos/evmos/v10/x/ibc/evm/types"
```
Comment (danburck) [inline app/app.go:136]:
```suggestion
	ibcevmtypes "github.com/evmos/evmos/v10/x/ibc/evm/types"
```

File: app/app.go [hunk 4]
```diff
@@ -180,7 +186,7 @@ var (
 		capability.AppModuleBasic{},
 		staking.AppModuleBasic{},
 		distr.AppModuleBasic{},
-		gov.NewAppModuleBasic(
+		gov.NewAppModuleBasic( // TODO add the new module's (ibc-evm-tx) handler when ready
```
Comment (danburck) [inline app/app.go:189]: @GAtom22 I can't find any ibc evm module specific governance proposals in the spec. Are you planning to add some? If not, then we can remove this todo

File: app/app.go [hunk 5]
```diff
@@ -223,6 +230,7 @@ var (
 		erc20types.ModuleName:          {authtypes.Minter, authtypes.Burner},
 		claimstypes.ModuleName:         nil,
 		incentivestypes.ModuleName:     {authtypes.Minter, authtypes.Burner},
+		ibcevmtxtypes.ModuleName:       {authtypes.Minter, authtypes.Burner},
```
Comment (danburck) [inline app/app.go:233]: @GAtom22 What do we need the minter and burner permissions for?

File: app/app.go [hunk 6]
```diff
@@ -290,6 +298,7 @@ type Evmos struct {
 	VestingKeeper    vestingkeeper.Keeper
 	RecoveryKeeper   *recoverykeeper.Keeper
 	RevenueKeeper    revenuekeeper.Keeper
+	IbcEvmTxKeeper   ibcevmtxkeeper.Keeper
```
Comment (danburck) [inline app/app.go:301]: @GAtom22 In what scenario does the ibc evm module account need to receive tokens and hold a balance? 

also, nit
```suggestion
	IBCEvmKeeper   ibcevmkeeper.Keeper
```

File: app/app.go [hunk 7]
```diff
@@ -374,6 +383,7 @@ func NewEvmos(
 
 	scopedIBCKeeper := app.CapabilityKeeper.ScopeToModule(ibchost.ModuleName)
 	scopedTransferKeeper := app.CapabilityKeeper.ScopeToModule(ibctransfertypes.ModuleName)
+	scopedIbcEvmTxKeeper := app.CapabilityKeeper.ScopeToModule(ibcevmtxtypes.ModuleName)
```
Comment (danburck) [inline app/app.go:386]:
```suggestion
	scopedIBCEvmKeeper := app.CapabilityKeeper.ScopeToModule(ibcevmtypes.ModuleName)
```

File: app/app.go [hunk 8]
```diff
@@ -565,6 +575,20 @@ func NewEvmos(
...
 
+	// create ibc evm tx keeper
+	app.IbcEvmTxKeeper = ibcevmtxkeeper.NewKeeper(
+		keys[ibcevmtxtypes.StoreKey],
+		appCodec,
+		app.GetSubspace(ibcevmtxtypes.ModuleName),
+		app.BankKeeper,
+		app.TransferKeeper,
+		*app.EvmKeeper,
+		nil,
```
Comment (danburck) [inline app/app.go:586]: why are we passing nil here?

File: app/app.go [hunk 9]
```diff
@@ -565,6 +575,20 @@ func NewEvmos(
 	// If evidence needs to be handled for the app, set routes in router here and seal
 	app.EvidenceKeeper = *evidenceKeeper
 
+	// create ibc evm tx keeper
+	app.IbcEvmTxKeeper = ibcevmtxkeeper.NewKeeper(
```
Comment (danburck) [inline app/app.go:579]: move to evmos module keepers, line 522

**Review by danburck — CHANGES_REQUESTED** [review]
@GAtom22 left some comments

**Pull request merged**